### PR TITLE
feat/legacy transactions

### PR DESCRIPTION
**Pull request by leozera14**
feat: Add SendLegacyTransaction component to handle Legacy and EIP-1559 transactions from dApps.

### CODE REVIEW

File: source/utils/ethUtil.ts [hunk 1]
```diff
@@ -26,6 +26,17 @@ export const decodeTransactionData = (params: ITransactionParams) => {
 
       return decoderValue;
     }
+
+    if (!data && value) {
+      const emptyDecoderObject = {
+        method: 'Contract Interaction',
```
Comment (claudiovb) [inline source/utils/ethUtil.ts:32]: The method name for transactions where there is no data but a value included should be send

File: source/utils/ethUtil.ts [hunk 2]
```diff
@@ -26,6 +26,17 @@ export const decodeTransactionData = (params: ITransactionParams) => {
 
       return decoderValue;
     }
+
+    if (!data && value) {
```
Comment (claudiovb) [inline source/utils/ethUtil.ts:30]: Check if the value is 0 or '0x0'if this if would fail

Comment (claudiovb) [reply source/utils/ethUtil.ts:30]: If this fails we should add some extra checking

File: source/scripts/Provider/EthProvider.ts [hunk 3]
```diff
@@ -27,6 +27,19 @@ export const EthProvider = (host: string) => {
 
     if (!decodedTx) throw cleanErrorStack(ethErrors.rpc.internal());
 
+    console.log('params', params);
+
+    if ((tx.type && tx.type === '0x0') || (!tx.type && !tx.data)) {
```
Comment (claudiovb) [inline source/scripts/Provider/EthProvider.ts:32]: Remove the tx type check here as we use this method for any sendTransaction case

File: source/scripts/Provider/EthProvider.ts [hunk 4]
```diff
@@ -27,6 +27,19 @@ export const EthProvider = (host: string) => {
 
     if (!decodedTx) throw cleanErrorStack(ethErrors.rpc.internal());
 
+    console.log('params', params);
```
Comment (claudiovb) [inline source/scripts/Provider/EthProvider.ts:30]: Before we merge it remember removing the  console logs that are strict for testing

**Review by claudiovb — CHANGES_REQUESTED** [review]
First review, still will add a commit with the fee and some packages fixes

File: source/utils/fetchGasAndDecodeFunction.ts [hunk 5]
```diff
@@ -16,21 +16,26 @@ export const fetchGasAndDecodeFunction = async (
   const { maxFeePerGas, maxPriorityFeePerGas } =
     await txs.getFeeDataWithDynamicMaxPriorityFeePerGas(); // this details maxFeePerGas and maxPriorityFeePerGas need to be passed as an option
   const nonce = await txs.getRecommendedNonce(dataTx.from); // This also need possibility for customization
+  //TODO: test approve, sendTransaction for contract interactions and other components to check if its working aproppriately
```
Comment (claudiovb) [inline source/utils/fetchGasAndDecodeFunction.ts:19]: Was this tested ?

**Review by claudiovb — CHANGES_REQUESTED** [review]
Overall it is looking really good, all the changes are working properly testing on the e2e test Dapp. But there are some small fixes commented that are still needed also removing all the git warnings from the files before we can merge this PR. And there is a TODO as well that needs to be addressed regarding testing the other methods (Aprovve, SendTransaction - contract interactions) like on Pegasys, Luxy, Uniswap, and other methods from test Dapp as well so we can be assured no new bugs were brought by this changes.

File: source/pages/Send/EditPriorityModal.tsx [hunk 6]
```diff
@@ -38,52 +66,91 @@ export const EditPriorityModal = (props: any) => {
...
-          'maxFeePerGas',
-          removeScientificNotation(0.8 * maxFeePerGas)
-        );
-        setCustomFee((prevState) => ({
-          ...prevState,
-          isCustom: true,
-          maxPriorityFeePerGas: 0.8 * maxPriorityFeePerGas,
-          maxFeePerGas: 0.8 * maxFeePerGas,
-        }));
+        if (!isSendLegacyTransaction) {
```
[the remark this answers is not in the document]
Comment (leozera14) [reply source/pages/Send/EditPriorityModal.tsx:70]: Fixed

File: source/utils/ethUtil.ts [hunk 7]
```diff
@@ -26,6 +26,22 @@ export const decodeTransactionData = (params: ITransactionParams) => {
 
       return decoderValue;
     }
+
+    const transactionValueValidation = [value === 0, String(value) === '0x0'];
```
[the remark this answers is not in the document]
Comment (leozera14) [reply source/utils/ethUtil.ts]: Fixed

File: source/pages/Send/EditPriorityModal.tsx [hunk 8]
```diff
@@ -160,80 +228,123 @@ export const EditPriorityModal = (props: any) => {
...
+                    onChange={(e) =>
+                      setCustomFee((prevState) => ({
+                        ...prevState,
+                        isCustom: true,
+                        gasPrice: +e.target.value,
+                      }))
+                    }
+                  />
+                </Form.Item>
+              </div>
```
[the remark this answers is not in the document]
Comment (leozera14) [reply source/pages/Send/EditPriorityModal.tsx:378]: Fixed

File: source/pages/Send/EditPriorityModal.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (leozera14) [reply source/pages/Send/EditPriorityModal.tsx:70]: Fixed

File: package.json [hunk 9]
```diff
@@ -187,4 +187,4 @@
   },
   "homepage": "https://github.com/syscoin/pali_wallet#readme",
   "author": "pollum labs"
-}
+}
```
[the remark this answers is not in the document]
Comment (leozera14) [reply package.json]: Fixed

**Review by claudiovb — APPROVED** [review]

**Pull request merged**